fruitCollection.js

```diff
@@ -0,0 +1,115 @@
+/*
+monkey eating fruit
+tree to tree
+find collection of tree of fruit
+
+list of tree
+each tree: x-coordinate, y-coordinate, num of fruit, vine length
+
+i: [2,4,5,10]
+o: max of fruit 
+
+const trees = [
+    [2, 4, 2, 6],
+    [10, 4, 5, 5],
+    [3, 8, 3, 6],
+    [12, 8, 6, 5],
+    ,[1, 6 2, 6],
+];
+
+*/
+
+const getDistanceBetweenNodes = (nodeOneCoordinates, nodeTwoCoordinates) => {
+    const [nodeOneXcoordinate, nodeOneYCoordinate, nodeOneFruitNum, nodeOneLengthDistance] =
+        nodeOneCoordinates;
+    const [nodeTwoXcoordinate, nodeTwoYCoordinate, nodeTwoFruitNum, nodeTwoLengthDistance] =
+        nodeTwoCoordinates;
+
+    const distance = Math.sqrt(
+        Math.pow(nodeTwoXcoordinate - nodeOneXcoordinate, 2) +
+            Math.pow(nodeTwoYCoordinate - nodeOneYCoordinate, 2),
+    );
+
+    const smallestDistanceBetweenTwoNodes = Math.min(nodeOneLengthDistance, nodeTwoLengthDistance);
+    return distance <= smallestDistanceBetweenTwoNodes;
+};
+
+const areConnected = (nodeOne, nodeTwo) => {
```

Comment:
This function doesn't return anything.

Are you sure your code works for this problem?